pkg/tfbridge/names.go

```diff
@@ -196,11 +199,25 @@ func FromName(options AutoNameOptions) func(res *PulumiResource) (interface{}, e
 			vs = options.Transform(vs)
 		}
 		if options.Randlen > 0 {
-			uniqueHex, err := resource.NewUniqueHex(vs+options.Separator, options.Randlen, options.Maxlen)
-			if err != nil {
-				return uniqueHex, errors.Wrapf(err, "could not make instance of '%v'", res.URN.Type())
+			// If Randchars is nil use hex chars
```

Comment:
We probably want to move this logic to pulumi/pulumi, so this PR is mostly around the thoughts on adding "Randchars" to the options struct.

Reply:
Got a branch to add a `NewUniqueName(urn URN, sequenceNumber int, prefix string, randlen, maxlen int, charset []rune)` to pulumi/pulumi, but it's fine for this to go in as is and switch to that later. Main thing is to start unblocking dependent providers so they don't have to call `From`

Reply:
Can we change this comment so that it explains why we're doing this? The code is simple enough to reason about.